### PR TITLE
Create loom-pointer-compression.md

### PR DESCRIPTION
**Pull request by thelifeandtimes**

### CODE REVIEW

**Comment by thelifeandtimes** [comment]
needs metadata tags and some editing.

File: content/grants/loom-pointer-compression.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+# Loom Pointer Compression Apprenticeship
+
+This project makes it possible to at least double the maximum addressible loom size, taking advantage of object allocation alignment to make use of one-or-more additional bits in our 29-bit (30-bits possible) loom-offset references.
+
+## Deliverables
+
+- an updated allocator (`u3a_walloc()`) that ensures at least 8-byte (2-word) alignment for all allocations
+- an updated de/reference api that compresses pointer-to-offset (`u3a_to_off()`) and decompresses offset-to-pointer (`u3a_to_ptr()`)
+- a migration from the old to new heap layout (`u3m_pack()`, but enforcing new alignment invariants)
```
Comment (joemfb) [inline content/grants/loom-pointer-compression.md]: I made some mistakes and was overly vague in this section:

- an updated allocator (`u3a_walloc()` and friends)
- pointer-to-offset (`u3a_outa()`)
- offset-to-pointer (`u3a_into()`)
- migration from the old to new heap layout (like `u3m_pack()`, but capable of reordering the heap to adjust minimum allocation alignment)

**Review by joemfb — DISMISSED** [review]
Should I push these changes onto this branch?

File: content/grants/loom-pointer-compression.md (hunk 1 above)
Comment (thelifeandtimes) [reply content/grants/loom-pointer-compression.md]: yes, please push whatever changes directly to this branch

Comment (thelifeandtimes) [reply content/grants/loom-pointer-compression.md]: @joemfb bump. please push any of your desired changes directly to this branch.

**Review by mikeosborne — APPROVED** [review]

**Comment by thelifeandtimes** [comment]
@mcevoypeter can you make necessary changes/approve?

**Review by mcevoypeter — APPROVED** [review]
I made the description more explicit. LGTM now.

**Review by mikeosborne — APPROVED** [review]

**Comment by joemfb** [comment]
@mcevoypeter your changes were correct, but i'm being intentionally vague about some of those bottom line results in this public document. I've added more details to the description to compensate.

@thelifeandtimes I also made the changes I referenced earlier. Sorry for the delay.

**Review by mikeosborne — APPROVED** [review]

**Pull request merged**